lib/puma/client.rb

```diff
@@ -98,6 +98,8 @@ def initialize(io, env=nil)
       @body_remain = 0
 
       @in_last_chunk = false
+
+      @read_buffer = +""
```

Comment:
What is this? I haven't seen this syntax before

Reply:
It is the short syntax for a non frozen String (if strings was to be frozen by default). Same as `String.new` which would read better IMHO.

Reply:
It also has better performance.

```
Benchmark.bm do |bm|
  bm.report { 100000000.times { String.new } }
  bm.report { 100000000.times { +'' } }
end
       user     system      total        real
   4.359000   0.000000   4.359000 ( 10.255661)
   2.954000   0.000000   2.954000 (  6.286254)
```

Reply:
I originally had `String.new` but this triggered a Rubocop `Performance/UnfreezeString` error.

I like `String.new` because it's intention is clear and we only allocate once. But I also see the benefit of using `+""` as a rule app-wide because performance is important in Puma. There are some places where this gain would be meaningful.

Reply:
Thanks, it's obviously not very easily googleable 😆